src/debug-generator/debug-generator.c

```diff
@@ -15,25 +15,46 @@
 #include "proc-cmdline.h"
 #include "recurse-dir.h"
 #include "special.h"
+#include "string-table.h"
 #include "string-util.h"
 #include "strv.h"
 #include "unit-file.h"
 #include "unit-name.h"
 
+typedef enum BootBreakpointType {
+        BOOT_BREAKPOINT_PRE_UDEV,
+        BOOT_BREAKPOINT_PRE_SYSROOT_MOUNT,
+        BOOT_BREAKPOINT_PRE_SWITCH_ROOT,
+        _BOOT_BREAKPOINT_TYPE_MAX,
+        _BOOT_BREAKPOINT_TYPE_INVALID = -EINVAL,
+} BootBreakpointType;
+
 static const char *arg_dest = NULL;
 static char *arg_default_unit = NULL;
 static char **arg_mask = NULL;
 static char **arg_wants = NULL;
 static bool arg_debug_shell = false;
 static char *arg_debug_tty = NULL;
 static char *arg_default_debug_tty = NULL;
+static BootBreakpointType arg_boot_breakpoint = _BOOT_BREAKPOINT_TYPE_INVALID;
 
 STATIC_DESTRUCTOR_REGISTER(arg_default_unit, freep);
 STATIC_DESTRUCTOR_REGISTER(arg_mask, strv_freep);
 STATIC_DESTRUCTOR_REGISTER(arg_wants, strv_freep);
 STATIC_DESTRUCTOR_REGISTER(arg_debug_tty, freep);
 STATIC_DESTRUCTOR_REGISTER(arg_default_debug_tty, freep);
 
+static const char* const boot_breakpoint_type_table[_BOOT_BREAKPOINT_TYPE_MAX] = {
```

Comment:
Maybe better to make the table also contain if each switch is supported in initrd and after switching root.